concept/type/owns.feature

```diff
@@ -0,0 +1,3952 @@
+    When transaction commits
+    When connection open schema transaction for database: typedb
+    Then <root-type>(<subtype-name>) set owns: reference[]
+    Then <root-type>(<subtype-name>) get owns(reference[]) set override: reference[]; fails
+    Then <root-type>(<subtype-name>) get owns(reference[]) set override: username[]; fails
+    Examples:
+      | root-type | supertype-name | subtype-name | value-type |
+      | entity    | person         | customer     | double     |
+      | relation  | description    | registration | string     |
+
+  Scenario Outline: <root-type> types cannot unset not owned list of ownerships
+    When put attribute type: username
+    When attribute(username) set value-type: <value-type>
+    When put attribute type: reference
+    When attribute(reference) set value-type: <value-type>
+    When <root-type>(<type-name>) set owns: username[]
+    Then <root-type>(<type-name>) get owns contain: username[]
+    Then <root-type>(<type-name>) unset owns: reference[]; fails
+    When transaction commits
+    When connection open schema transaction for database: typedb
+    Then <root-type>(<type-name>) get owns contain: username[]
+    Then <root-type>(<type-name>) unset owns: reference[]; fails
+    Then <root-type>(<type-name>) unset owns: username[]
+    Then <root-type>(<type-name>) unset owns: username[]; fails
+    Then <root-type>(<type-name>) get owns is empty
+    When transaction commits
+    When connection open read transaction for database: typedb
+    Then <root-type>(<type-name>) get owns is empty
+    Examples:
+      | root-type | type-name   | value-type |
+      | entity    | person      | string     |
+      | relation  | description | long       |
+
+  Scenario Outline: <root-type> types cannot unset inherited ownership
+    When put attribute type: username
+    When attribute(username) set value-type: <value-type>
+    When <root-type>(<supertype-name>) set owns: username[]
+    Then <root-type>(<supertype-name>) get owns contain: username[]
+    Then <root-type>(<subtype-name>) get owns contain: username[]
+    Then <root-type>(<subtype-name>) unset owns: username[]; fails
+    When transaction commits
+    When connection open schema transaction for database: typedb
+    Then <root-type>(<subtype-name>) get owns contain: username[]
+    Then <root-type>(<subtype-name>) unset owns: username[]; fails
+    Examples:
+      | root-type | supertype-name | subtype-name | value-type |
+      | entity    | person         | customer     | string     |
+      | relation  | description    | registration | long       |
+
+  Scenario Outline: <root-type> types cannot set lists of attributes alongside scalar attribute
+    When put attribute type: username
+    When attribute(username) set value-type: <value-type>
+    When <root-type>(<type-name>) set owns: username
+    Then <root-type>(<type-name>) get owns contain: username
+    # TODO: Or it will override the non-list definition?
```

Comment:
Some of these questions as well.

Reply:
Hmm ok, maybe we use a simpler approach of they have to undefine first, then define?
Although going from owns -> owns[] could be a no-op in many cases where only 1 ownership exists...
but i think is more general to make them delete and re-insert somehow, eg make a new ownership, then undefine and delete the old one.

Reply:
My idea was that we need to unset the old value (scalar or list) and then set the new value (list or scalar). I guess I'll add a test for it.

Reply:
But I may end up refactoring it based on this message https://discord.com/channels/665254494820368395/699946888572567611/1250453567903502517

I don't have a final decision yet, I may skip this refactoring for now...